LoRaEngine/test/LoRaWan.IntegrationTest/C2DMessageTest.cs

```diff
@@ -563,5 +559,200 @@ public async Task C2D_Using_Invalid_FPort_Should_Be_Ignored(byte fport, string d
+                        (messageBody) => {
+                            return messageBody.StartsWith(expectedUDPMessageV1) || messageBody.StartsWith(expectedUDPMessageV2);
+                        },
+                        new SearchLogOptions
+                        {
+                            Description = $"{expectedUDPMessageV1} or {expectedUDPMessageV2}",
+                            MaxAttempts = 1,
+                        });
+
+                    // We should only receive the message once
+                    if (searchResults.Found)
+                    {
+                        this.Log($"{device.DeviceID}: Found C2D message in log (after sending {i}/10)");
+                        foundC2DMessage = true;
+                    }
+                }
+
+                if (!foundReceivePacket)
+                {
+                    if (this.ArduinoDevice.SerialLogs.Contains(expectedRxSerial1))
+                    {
+                        this.Log($"{device.DeviceID}: Found serial log message {expectedRxSerial1} in log (after sending {i}/10)");
+                        foundReceivePacket = true;
+                    }
+                }
+
+                if (!foundReceivePacketInRX2)
+                {
+                    if (this.ArduinoDevice.SerialLogs.Any(x => x.StartsWith(expectedRxSerial2)))
+                    {
+                        this.Log($"{device.DeviceID}: Found serial log message {expectedRxSerial2} in log (after sending {i}/10)");
+                        foundReceivePacketInRX2 = true;
+                    }
+                }
+
+                if (foundReceivePacket && foundReceivePacketInRX2 && foundC2DMessage)
+                {
+                    this.Log($"{device.DeviceID}: Found all messages in log (after sending {i}/10)");
+                    break;
+                }
+
+                this.TestFixture.ClearLogs();
+
+                await Task.Delay(Constants.DELAY_BETWEEN_MESSAGES);
+            }
+
+            Assert.True(foundC2DMessage, $"Did not find {expectedUDPMessageV1} or {expectedUDPMessageV2} in logs");
+
+            // checks if serial received the message
+            if (!foundReceivePacket)
+            {
+                foundReceivePacket = this.ArduinoDevice.SerialLogs.Contains(expectedRxSerial1);
+            }
+
+            Assert.True(foundReceivePacket, $"Could not find lora receiving message '{expectedRxSerial1}'");
+
+
+            // checks if serial received the message in RX2
+            if (!foundReceivePacketInRX2)
+            {
+                foundReceivePacketInRX2 = this.ArduinoDevice.SerialLogs.Any(x => x.StartsWith(expectedRxSerial2));
+            }
+
+            Assert.True(foundReceivePacketInRX2, $"Could not find lora receiving message '{expectedRxSerial2}'");
+        }
+
+        // [Fact]
+        // public async Task C2D_With_PreferredWindow2_Should_Process_C2d_In_First_Upstream()
```

Comment:
Is this test needed for later stage or is this just a legacy item?

Reply:
It is the test I used to build the feature. I will remove it.